popup.js

```diff
@@ -0,0 +1,251 @@
+                div.className = 'noFixes';
+                div.innerHTML = 'No high impact suggestions. Bueno!';
+                document.body.insertBefore(div, null);
+            }
+        };
+
+        // Helper function that sorts results in order of impact.
+        function sortByImpact(a, b) {
+            return b.impact - a.impact;
+        }
+
+
+
+
+        //PIE CHART
+        var RESOURCE_TYPE_INFO = [
+            { label: 'JavaScript', field: 'javascriptResponseBytes', color: 'e2192c' },
+            { label: 'Images', field: 'imageResponseBytes', color: 'f3ed4a' },
+            { label: 'CSS', field: 'cssResponseBytes', color: 'ff7008' },
+            { label: 'HTML', field: 'htmlResponseBytes', color: '43c121' },
+            { label: 'Flash', field: 'flashResponseBytes', color: 'f8ce44' },
+            { label: 'Text', field: 'textResponseBytes', color: 'ad6bc5' },
+            { label: 'Other', field: 'otherResponseBytes', color: '1051e8' },
+        ];
+
+        function displayResourceSizeBreakdown(result) {
+            var stats = result.pageStats;
+            var labels = [];
+            var data = [];
+            var colors = [];
+            var totalBytes = 0;
+            var largestSingleCategory = 0;
+            for (var i = 0, len = RESOURCE_TYPE_INFO.length; i < len; ++i) {
+                var label = RESOURCE_TYPE_INFO[i].label;
+                var field = RESOURCE_TYPE_INFO[i].field;
+                var color = RESOURCE_TYPE_INFO[i].color;
+                if (field in stats) {
+                    var val = Number(stats[field]);
+                    totalBytes += val;
+                    if (val > largestSingleCategory) largestSingleCategory = val;
+                    labels.push(label);
+                    data.push(val);
+                    colors.push(color);
+                }
+            }
+            // Construct the query to send to the Google Chart Tools.
+            var query = [
+                'chs=300x140',
+                'cht=p3',
+                'chts=' + ['000000', 16].join(','),
+                'chco=' + colors.join('|'),
+                'chd=t:' + data.join(','),
+                'chdl=' + labels.join('|'),
+                'chdls=000000,14',
+                'chp=1.6',
+                'chds=0,' + largestSingleCategory,
+            ].join('&');
+            var i = document.createElement('img');
+            var be4 = document.getElementById('analysisChart');
+            i.src = 'http://chart.apis.google.com/chart?' + query;
+            document.body.insertBefore(i, be4);
+        };
+
+
+
+
+        function runPagespeed() {
+            var queryParams = {
+                url: URL_TO_GET_RESULTS_FOR,
+                key: APIKEY,
+                screenshot: true
+            }
+            axios.get(API_URL, { params: queryParams }) //
+                .then(function(response) {
+                    console.log('response object', response);
```

Comment:
you shouldn't leave any console.logs in your files ... they're only meant for debugging and should be removed afterwards .. please erase this line